packages/PowerGrid/src/fit/fit-data.ts

```diff
@@ -0,0 +1,301 @@
+ * the property is not defined in target and is defined in source. */
+function mergeProperties(properties: Property[], source: any, target: any) {
+  if (!source || !target)
+    return;
+
+  for (let p of properties)
+    if (p.name !in target && p.name in source)
+      p.set(target, p.get(source));
+}
+
+
+function createDefaultChartData(): IFitChartData {
+  return {
+    chartOptions: createFromProperties(fitChartDataProperties),
+    seriesOptions: createFromProperties(fitSeriesProperties)
+  };
+}
+
+
+/** Returns existing, or creates new column default chart options. */
+export function getColumnChartOptions(gridColumn: DG.GridColumn): IFitChartData {
+  return gridColumn.tags[TAG_FIT] ??= createDefaultChartData();
+}
+
+
+export function getChartBounds(chartData: IFitChartData): DG.Rect {
+  let o = chartData.chartOptions;
+  if (o?.minX && o.minY && o.maxX && o.maxY)
+    return new DG.Rect(o.minX, o.minY, o.maxX - o.minX, o.maxY - o.minY);
+  if (!chartData.series?.length || chartData.series.length == 0)
+    return new DG.Rect(0, 0, 1, 1);
+  else {
+    let bounds = getDataBounds(chartData.series[0].points);
+    for (let i = 1; i < chartData.series!.length; i++)
+      bounds = union(bounds, getDataBounds(chartData.series[i].points));
+    return bounds;
+  }
+}
+
+
+//TODO: move to DG.Rect
+export function getDataBounds(points: IFitPoint[]): DG.Rect {
+  let minX = points[0].x;
+  let minY = points[0].y;
+  let maxX = points[0].x;
+  let maxY = points[0].y;
+
+  for (let i = 1; i < points.length; i++) {
+    minX = Math.min(minX, points[i].x);
+    minY = Math.min(minY, points[i].y);
+    maxX = Math.max(maxX, points[i].x);
+    maxY = Math.max(maxY, points[i].y);
+  }
+
+  return new DG.Rect(minX, minY, maxX - minX, maxY - minY);
+}
+
+
+//TODO: move to DG.Rect
+function fromPoints(x1: number, y1: number, x2: number, y2: number): DG.Rect {
+  let minX = Math.min(x1, x2);
+  let minY = Math.min(y1, y2);
+  return new DG.Rect(minX, minY, Math.max(x1, x2) - minX, Math.max(y1, y2) - minY)
+}
+
+
+//TODO: move to DG.Rect
+function union(r1: DG.Rect, r2: DG.Rect): DG.Rect {
+  return fromPoints(
+    Math.min(r1.left, r2.left),
+    Math.min(r1.top, r2.top),
+    Math.max(r1.right, r2.right),
+    Math.max(r1.bottom, r2.bottom));
+}
+
+
+/** Fits the series data according to the series fitting settings */
+export function fitSeries(series: IFitSeries): FitResult {
+
+  //TODO: optimize the calculation of the initial parameters
+  const dataBounds = getDataBounds(series.points);
+  let ys = series.points.map(p => p.y);
+  let medY = ys[Math.floor(series.points.length / 2)];
+  let xAtMedY = -1;
+  for (let i = 0; i < series.points.length; i++) {
+    if (series.points[i].y === medY) {
+      xAtMedY = series.points[i].x;
+      break;
+    }
+  }
+  let initialParams = [dataBounds.bottom, 1.2, xAtMedY, dataBounds.top];
+
+  return fit(
+    {x: series.points.map(p => p.x), y: series.points.map(p => p.x)},
```

Comment:
`y` is set the same as `x`. Is it intended?

Reply:
i'll look at it